test/ambiguous.jl

```diff
@@ -135,7 +135,7 @@ ambs = detect_ambiguities(Ambig5)
 # Test that Core and Base are free of ambiguities
 # TODO jb/subtype: we now detect a lot more
 @test !isempty(detect_ambiguities(Core, Base; imported=true, ambiguous_bottom=true))
```

Comment:
should maybe have a comment here that this being non-empty isn't actually required, but is likely to be the case